//
// Copyright 2019 The Project Oak Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto3";

package oak.examples.abitest;

message ABITestRequest {
  // Regular expression of test names to include in the test run.
  string include = 1;
  // If non-empty, holds a regular expression of test names to exclude from the
  // test run.  Tests are only run if they match include and do not match
  // exclude.
  string exclude = 2;
}

message ABITestResponse {
  message TestResult {
    string name = 1;
    bool success = 2;
    string details = 3;
    bool disabled = 4;
  }
  repeated TestResult results = 1;
}

service OakABITestService {
  rpc RunTests(ABITestRequest) returns (ABITestResponse);
}
